lib/setupHooks/installFromCargoBuildLogHook.sh

```diff
@@ -32,12 +38,11 @@ function installFromCargoBuildLog() (
     rmdir --ignore-fail-on-non-empty "${loc}"
   }
 
-  echo "${logs}" | jq -r "${select_bins}" | installArtifacts "${dest}/bin"
-
   command cargo metadata --format-version 1 | jq '.workspace_members[]' | (
     while IFS= read -r ws_member; do
-      local select_member_libs="select(.package_id == ${ws_member}) | ${select_lib_files}"
-      echo "${logs}" | jq -r "${select_member_libs}" | installArtifacts "${dest}/lib"
+      local select_member="select(.package_id == ${ws_member})"
+      echo "${logs}" | jq -r "${select_member} | ${select_lib_files}" | installArtifacts "${dest}/lib"
+      echo "${logs}" | jq -r "${select_member} | ${select_bins}" | installArtifacts "${dest}/bin"
```

Comment:
I think doing things this way will attempt to install _all binaries_ for a particular member, but that doesn't necessarily mean we have built them all. Consider if the derivation is built with `cargo build -p myCrate --bin foo` but the crate specifies `foo` and `bar` binaries. Would this check also attempt to install `bar`?

Perhaps we should be installing all binaries that were built _and are also not considered dependencies_?

Reply:
Yeah it would definitely be easier to manage (and test!) with a Rust program. We already have `crane-utils` in the repo (currently only used for vendoring git-dependencies). Feel free to add a new binary in there which would process a cargo installation log!

Reply:
Taking another look, this is actually already using `'"${select_non_dep}"` above, doesn't that cover the issue here?

Reply:
Looking back at my original comment, I think it might have come out of me misunderstanding the implementation so I'll mark this as resolved!